### PR TITLE
Added option to wdigest module to check registry key without editing it

### PR DESCRIPTION
**Pull request by probird5**
Added this option since I needed it on a pentest where the client wanted a list of hosts with wdigest enabled. Did not want to modify or add existing key.

vmware_EQ1tgrJNa9

### CODE REVIEW

**Comment by NeffIsBack** [comment]
There were some formatting changes with single and double quotation marks that would be reverted by your PR. Can you check that?😊

**Comment by probird5** [comment]
> There were some formatting changes with single and double quotation marks that would be reverted by your PR. Can you check that?😊

Sorry about that, should be fixed now.

File: cme/modules/wdigest.py [hunk 1]
```diff
@@ -113,3 +115,30 @@ def wdigest_disable(self, context, smbconnection):
...
+
+            try:
+                rtype, data = rrp.hBaseRegQueryValue(remoteOps._RemoteOperations__rrp, keyHandle, "UseLogonCredential\x00")
+                if int(data) == 1:
+                    context.log.success("UseLogonCredential registry key is enabled")
+                else:
+                    context.log.error("Unexpected registry value for UseLogonCredential: %s" % data)
+            except DCERPCException as d:
+                if "winreg.HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\Control\\SecurityProviders\\WDigest" in str(d):
+                    context.log.error("UseLogonCredential registry key is disabled (registry key not found)")
```
Comment (mpgn) [inline cme/modules/wdigest.py:138]: can you update to context.log.fail(...) :)

File: cme/modules/wdigest.py [hunk 2]
```diff
@@ -113,3 +115,30 @@ def wdigest_disable(self, context, smbconnection):
...
+                else:
+                    context.log.error("UseLogonCredential registry key not present")
```
Comment (mpgn) [inline cme/modules/wdigest.py:140]: can you update to context.log.fail(...) :)

**Review by mpgn — CHANGES_REQUESTED** [review]

**Comment by mpgn** [comment]
Can you also add the "check" on the option list:

https://github.com/mpgn/CrackMapExec/pull/4/files#diff-d259610f4607f5b77a14ce1b3cc61f0820e107c1fc564a8cf2b73afe446bf274R19

File: cme/modules/wdigest.py (hunk 1 above)
Comment (probird5) [reply cme/modules/wdigest.py]: Done :)

File: cme/modules/wdigest.py (hunk 2 above)
Comment (probird5) [reply cme/modules/wdigest.py]: Done :)

**Review by mpgn — APPROVED** [review]

**Pull request merged**